devito/arch/archinfo.py

```diff
@@ -376,6 +377,35 @@ def get_cuda_path():
     return None
 
 
+@memoized_func
+def get_m1_llvm_path():
+    # Check if Apple's llvm is installed (installable via Homebrew), which supports
+    # OpenMP.
+    # Check that we are on apple system
+    if sys.platform != 'darwin':
+        raise ValueError('Apple LLVM is only available on Mac OS X.')
+    # *** First check if LLVM is installed
+    ver = run(["clang", "--version"], stdout=PIPE, stderr=DEVNULL).stdout.decode("utf-8")
+    # *** Second check if this clang version targets arm64 (M1)
+    if "arm64-apple" in ver:
```

Comment:
Shouldn;t this be try except? How can you ensure that Apple's x86 clang found?

Reply:
Because apple system comes with clang as factory default the same way any linux comes with GCC 4.9.
This function also can only be hit from withing ClangCompiler